src/components/protocol_handler/include/protocol_handler/protocol_handler_impl.h

```diff
@@ -569,6 +620,8 @@ class ProtocolHandlerImpl
 #ifdef TELEMETRY_MONITOR
   PHTelemetryObserver* metric_observer_;
 #endif  // TELEMETRY_MONITOR
+  PTUState ptu_state_;
+  struct SessionInfo pending_session_;
```

Comment:
@istoilovgithub  Done in 3ae901fb0fb10922ff83ee7464da02f4df4b3540
Locking implemented